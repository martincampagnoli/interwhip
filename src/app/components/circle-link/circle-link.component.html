<a
  class="circle-link"
  title="{{ title() }}"
  href="{{ link() }}"
  target="_blank"
  rel="noopener"
>
  <img width="30" height="30" src="{{ img() }}" />
</a>
