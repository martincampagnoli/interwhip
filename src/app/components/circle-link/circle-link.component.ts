import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

/**
 * Component representing a circular link with an image and title.
 */
@Component({
    selector: 'app-circle-link',
    templateUrl: './circle-link.component.html',
    styleUrls: ['./circle-link.component.scss'],
    changeDetection: ChangeDetectionStrategy.OnPush,
    standalone: false
})
export class CircleLinkComponent {
  /**
   * The title of the circular link.
   */
  @Input() title: string | undefined;

  /**
   * The URL link associated with the circular link.
   */
  @Input() link: string | undefined;

  /**
   * The URL of the image to be displayed within the circular link.
   */
  @Input() img: string | undefined;

  /**
   * Constructs the CircleLinkComponent.
   */
  constructor() {}
}
