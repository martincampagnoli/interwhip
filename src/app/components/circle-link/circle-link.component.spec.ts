import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CircleLinkComponent } from './circle-link.component';

describe('CircleLinkComponent', () => {
  let component: CircleLinkComponent;
  let fixture: ComponentFixture<CircleLinkComponent>;
  let element: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [CircleLinkComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CircleLinkComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;

    fixture.componentRef.setInput('title', 'Test Title');
    fixture.componentRef.setInput('link', 'https://example.com');
    fixture.componentRef.setInput('img', 'test-image.jpg');

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should have the correct title', () => {
    const linkElement = element.querySelector('a');
    expect(linkElement?.getAttribute('title')).toBe('Test Title');
  });

  it('should have the correct link', () => {
    const linkElement = element.querySelector('a');
    expect(linkElement?.getAttribute('href')).toBe('https://example.com');
  });

  it('should have the correct image source', () => {
    const imgElement = element.querySelector('img');
    expect(imgElement?.getAttribute('src')).toBe('test-image.jpg');
  });

  it('should have target="_blank" and rel="noopener"', () => {
    const linkElement = element.querySelector('a');
    expect(linkElement?.getAttribute('target')).toBe('_blank');
    expect(linkElement?.getAttribute('rel')).toBe('noopener');
  });

  it('should match snapshot with default props', () => {
    expect(element).toMatchSnapshot();
  });

  it('should match snapshot with different props', () => {
    const newFixture = TestBed.createComponent(CircleLinkComponent);
    newFixture.componentRef.setInput('title', 'Documentation');
    newFixture.componentRef.setInput('link', 'https://angular.io');
    newFixture.componentRef.setInput('img', 'angular-logo.png');
    newFixture.detectChanges();
    expect(newFixture.nativeElement).toMatchSnapshot();
  });

  it('should match snapshot with special characters in title', () => {
    const newFixture = TestBed.createComponent(CircleLinkComponent);
    newFixture.componentRef.setInput('title', 'My Awesome App & More!');
    newFixture.componentRef.setInput(
      'link',
      'https://example.com/path?param=value'
    );
    newFixture.componentRef.setInput('img', 'icon-with-special-chars.svg');
    newFixture.detectChanges();
    expect(newFixture.nativeElement).toMatchSnapshot();
  });
});
