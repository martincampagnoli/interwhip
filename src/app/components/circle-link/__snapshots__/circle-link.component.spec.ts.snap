// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleLinkComponent should match snapshot with default props 1`] = `
<div
  id="root5"
>
  <a
    class="circle-link"
    href="https://example.com"
    rel="noopener"
    target="_blank"
    title="Test Title"
  >
    <img
      alt="icon"
      height="30"
      src="test-image.jpg"
      width="30"
    />
  </a>
</div>
`;

exports[`CircleLinkComponent should match snapshot with different props 1`] = `
<div
  id="root7"
>
  <a
    class="circle-link"
    href="https://angular.io"
    rel="noopener"
    target="_blank"
    title="Documentation"
  >
    <img
      alt="icon"
      height="30"
      src="angular-logo.png"
      width="30"
    />
  </a>
</div>
`;

exports[`CircleLinkComponent should match snapshot with special characters in title 1`] = `
<div
  id="root9"
>
  <a
    class="circle-link"
    href="https://example.com/path?param=value"
    rel="noopener"
    target="_blank"
    title="My Awesome App & More!"
  >
    <img
      alt="icon"
      height="30"
      src="icon-with-special-chars.svg"
      width="30"
    />
  </a>
</div>
`;
